### PR TITLE
add producedAnalyses to query; add logic to filter suppressed producedAnalyses

### PR DESCRIPTION
**Pull request by Buwujiu**

### CODE REVIEW

File: src/rdpc/analysesProcessor.ts [hunk 1]
```diff
@@ -206,13 +207,33 @@ export const getAllMergedDonor = async ({
...
   }
   return mergedDonors;
 };
 
+// Removes runs with suppressed producedAnalyses
+export const removeRunsWithSuppressedAnalyses = (
+  analyses: Analysis[]
+): Analysis[] => {
+  const result = analyses.reduce<Analysis[]>((acc, analysis) => {
+    const filteredRuns = analysis.runs.reduce<Run[]>((_acc, run) => {
```
Comment (joneubank) [inline src/rdpc/analysesProcessor.ts:223]: Don't be afraid to use longer names to make the different accumulators more readable. Maybe:
`resultAcc` and `runsAcc`

File: src/rdpc/fetchAnalyses.ts [hunk 2]
```diff
@@ -27,13 +27,19 @@ query($analysisFilter: AnalysisFilter, $analysisPage: Page, $workflowRepoUrl: St
...
-            analysisType
+          runId
+          state
+          repository
+          inputAnalyses {
+              analysisId
+              analysisType
           }
+          producedAnalyses
+            (filter: { analysisState: PUBLISHED}) {
```
Comment (joneubank) [inline src/rdpc/fetchAnalyses.ts:38]: 👍

File: src/rdpc/analysesProcessor.ts [hunk 3]
```diff
@@ -206,13 +207,33 @@ export const getAllMergedDonor = async ({
...
+      if (run.producedAnalyses && run.producedAnalyses.length !== 0) {
+        _acc.push(run);
+      }
+      return _acc;
+    }, []);
+
+    analysis.runs = filteredRuns;
```
Comment (joneubank) [inline src/rdpc/analysesProcessor.ts:230]: We shouldn't be modifying the query contents like this. Consider making a new object with the data we want to pass on and adding that to your result.

File: src/rdpc/analysesProcessor.ts (hunk 1 above)
Comment (joneubank) [reply src/rdpc/analysesProcessor.ts:223]: This just looks like a filter operation, can you simplify this block?
```
const filteredRuns = analysis.runs.filter(run => run.producedAnalyses && run.producedAnalyses.length > 0);
```

**Review by joneubank — CHANGES_REQUESTED** [review]

**Review by joneubank — APPROVED** [review]
Thanks for the updates! Good stuff here.

**Comment by Buwujiu** [comment]
version = 1.16.1

**Pull request merged**